src/recon_test/test_consistency_with_root.cxx

```diff
@@ -0,0 +1,402 @@
+  int num_failed_nonTOF_lor_events = 0;
+  double nonTOF_distance_threshold;
+
+  /// TOF VARIABLES
+  //! A copy of the scanner timing resolution in mm
+  const float failure_tolerance_TOF = 0.05;
+  std::vector<WeightedCoordinate> TOF_LOR_peak_value_coords;
+  double TOF_distance_threshold;
+  int num_failed_TOF_lor_events = 0;
+};
+
+void
+ROOTConsistencyTests::run_tests()
+{
+  // Loop over each of the ROOT files in the test_data directory
+  num_tests = 8;
+  test_results_nonTOF = std::vector<bool> (num_tests, true);
+  test_results_TOF = std::vector<bool> (num_tests, true);
+
+  cerr << "Testing the view offset consistency between GATE/ROOT and STIR. \n";
+  for (int i = 1; i <= num_tests; ++i)
+  {
+    test_index = i;
+    cerr << "\nTesting dataset " << std::to_string(test_index) << "...\n";
+
+    setup();
+    process_list_data();
+
+    cerr << "\nResults for dataset: " << std::to_string(test_index) << std::endl;
+    post_processing_nonTOF();
+    post_processing_TOF();
+  }
+
+  // Print results for easy readability
+  cerr << "\nTest Results\n"
+          "------------------------------------------\n"
+          "\tTest Index\t|\tnonTOF\t|\tTOF\n"
+          "------------------------------------------\n";
+  for (int j = 0; j < num_tests; ++j)
+    cerr << "\t\t" << std::to_string(j + 1) << "\t\t|\t"
+         << ((test_results_nonTOF[j]) ? "Pass" : "Fail") << "\t|\t"
+         << ((test_results_TOF[j]) ? "Pass" : "Fail") << std::endl;
+}
+
+void
+ROOTConsistencyTests::
+setup()
+{
+  // Create the point source (discretised_density_sptr) with GenerateImage from parameter file
+  // GenerateImage requires `const char* const par_filename`.
+  GenerateImage image_gen_application(get_generate_image_par_filename().c_str());
+  image_gen_application.compute();
+  discretised_density_sptr = image_gen_application.get_output_sptr();
+
+  // Needs to be cast to VoxelsOnCartesianGrid to be able to calculate the centre of gravity
+  const VoxelsOnCartesianGrid<float> &discretised_cartesian_grid =
+      dynamic_cast<VoxelsOnCartesianGrid<float> &>(*discretised_density_sptr);
+
+  // Find the center of mass and grid spacing of the original data
+  original_coords = find_centre_of_gravity_in_mm(discretised_cartesian_grid);
+  grid_spacing = discretised_cartesian_grid.get_grid_spacing();
+
+  // For post_processing and debugging, create a list of the closest LOR coords
+  nonTOF_closest_voxels_list.clear();
+  TOF_LOR_peak_value_coords.clear();
+
+  // Reset the number of processed and failed events
+  num_failed_nonTOF_lor_events = 0;
+  num_failed_TOF_lor_events = 0;
+  num_events_tested = 0;
+
+  // Failure conditioner and recording
+  nonTOF_distance_threshold = 1.5 * norm(grid_spacing); // Using norm(grid_spacing) as a nonTOF_distance_threshold
+  TOF_distance_threshold = 3.3 * norm(grid_spacing); // With the value 3.3*norm(grid_spacing), test_root_consistency passes for all sources
```

Comment:
This threshold has to depend on the TOF resolution. Computing it is hard, but we can now work our way backwards. Something like this:
```suggestion
  // set TOF distance threshold
  {
    // With the default files, we found that 3.3*norm(grid_spacing) is a reasonable limit.
    // We try to generalise this to other data (although it will likely need further
    // adjustment for cases with very different TOF resolution).
    const auto& pdi = *lm_data_sptr->get_proj_data_info_sptr();
    const float approx_tof_resolution =
      pdi.get_scanner()*get_timing_resolution()*pdf.get_tof_mash_factor();
    const float org_threshold = 3.3F / 75; // CHECK
    this->TOF_distance_threshold = approx_tof_resolution * org_threshold * norm(grid_spacing);
  }
```
or should `failure_tolerance_TOF` be used (according to its docstring, but I doubt it)

Reply:
obviously, at this point, we don't have `lm_data_sptr`, so this (or the `proj_data_info_sptr) will need to be passed to `set_up()`

Reply:
`get_tof_mash_factor()` doesnt exist. Im not 100% sure what it physically is either

Reply:
sure it does. https://github.com/NikEfth/STIR/blob/tof_sino_UCL/src/include/stir/ProjDataInfo.h#L243

TOF mash factor says how many of the "scanner" TOF bins are combined together (to make them larger)